mathics_django/web/media/js/graphics3d.js

```diff
@@ -9,8 +9,8 @@ const drawFunctions = {
 		return new THREE.Points(geometry, new THREE.ShaderMaterial({
 			transparent: true,
 			uniforms: {
-				// 84 is the scale for this size of canvas
-				size: { value: element.pointSize * 84 },
+				// 400 is the size of canvas
```

Comment:
I'd add a comment either here or on on line 318 of mathics.js noting that the two values are related. (Again it would be better is a global value were set for the "400" someplace and could be picked up both places.  Then no comment is needed linking the two pieces of code together.

Reply:
@rocky this makes sense, also there's a TODO for implement a changing size (e.g. for smaller screens), maybe just CSS with a rule of maximum width and width could help. As you said the points are small, the multiplier should be 400, a saw the points in WL documentation and 400 looks very similar.

Reply:
@rocky when drawGraphics3D called the div still wasn't added to the DOM, so get it's size isn't possible (I could get the CSS text and split it, but I guess this too hacky), maybe add an parameter with the width of the container to drawGraphics3D?

Also, about the performance it could be improved a bit if when `Cuboid` is called Mathics core ask to draw cubes, not polygons (8 of them). Also if it be done multiple cubes could be drawn at the same time, so I could use the same approximation that I used to improve the performance of drawing spheres.

I also think that the number of segments of the spheres is too high (we already have antialiasing), I'll check other number of segments other than (48, 48).

About the loading times, them could be reduced if we use just static content instead of the HTML templates. Also I was looking at the folder `scriptaculous` and some of the files don't look to be used.

Also, in the future, performance could be improved replacing Django by Flask.

Reply:
> @rocky this makes sense, also there's a TODO for implement a changing size (e.g. for smaller screens),

Ok. I totally understand that not everything can be done all at once and we'll have to iterate and iterate on things. So the only thing that is important is to acknowledge that there is work to be done and have it jotted down somewhere so we won't forget.

> maybe just CSS with a rule of maximum width and width could help. As you said the points are small, the multiplier should be 400, a saw the points in WL documentation and 400 looks very similar.

I don't know, but do whatever you think is right. Maybe @mmatera will chime in when he gets a chance. 

> 
> Also when drawGraphics3D is called the div still wasn't added to the DOM, so get it's size isn't possible (I could get the CSS text and split it, but I guess this too hacky), maybe add an parameter with the width of the container to drawGraphics3D?

That'd work. Beggars can't be choosers so do whatever you think best  (or leave if it is too difficult for now). 

> 
> About the performance:
> 
> 1. it could be improved a bit if when `Cuboid` is called Mathics core ask to draw cubes, not polygons (8 of them). Also if it be done multiple cubes could be drawn at the same time, so I could use the same approximation that I used to improve the performance of drawing spheres.

Ok. 

> 2. I think that the number of segments of the spheres is too high (we have antialiasing), I'll check other number of segments other than (48, 48).

Ok - thakns.

> 3. I was looking at the folder `scriptaculous` and some of the files don't look to be used (even one which is imported).

> 4. I [guess] vanilla JavaScript would be faster than `prototype.js`.

I guess testing here is in order. 

> 5. the loading times, them could be reduced if we use just static content instead of the HTML templates for some of the files (the base HTML could be split into two files: with_login.html and without_login.html), but I think we should focus on other things optimizations right now.
> 6. In the future, performance could be improved replacing Django by Flask.

As for replacing Django - the long-term plan is to ditch it for Jupyter. But right now it is useful for understanding how to fixing up the API since it is already there.